compiler/il/NodeExtension.hpp

```diff
@@ -58,8 +58,7 @@ class NodeExtension
     template <class T>
     T  setElem(uint16_t index, T elem)
        {
-       (T)(_data[index] = (uintptr_t)elem);
-       return (T)_data[index];
+       return (T)(_data[index] = (uintptr_t)elem);
```

Comment:
The first statement was causing a compiler warning because the cast on the first line was not being consumed by anything, as @Leonardo2718 stated.